src/core/include/units/reference.h

```diff
@@ -51,16 +50,17 @@ struct reference {
   static constexpr QuantitySpec auto quantity_spec = Q;
   static constexpr Dimension auto dimension = Q.dimension;
   static constexpr Unit auto unit = U;
+
+  template<RepresentationOf<Q.character> Rep>
+  // TODO can we somehow return an explicit quantity type here?
+  [[nodiscard]] constexpr Quantity auto operator()(Rep&& value) const
+  {
+    return quantity<reference{}, Rep>(std::forward<Rep>(value));
```

Comment:
Does `quantity<reference{}, Rep>` not work due to `reference` being incomplete? Consider `std::same_as<quantity<reference{}, Rep>>` if that works.

Reply:
It works, thanks!